src/Transition/CreateNewIDFUsingRulesV9_0_0.f90

```diff
@@ -442,7 +455,90 @@ SUBROUTINE CreateNewIDFUsingRules(EndOfFile,DiffOnly,InLfn,AskForInput,InputFile
               ! If your original object starts with P, insert the rules here
 
               ! If your original object starts with R, insert the rules here
-
+              CASE('RUNPERIOD')
+                CALL GetNewObjectDefInIDD(ObjectName,NwNumArgs,NwAorN,NwReqFld,NwObjMinFlds,NwFldNames,NwFldDefaults,NwFldUnits)
+                ! spend some time mining out the state of the run period object
+                RunPeriodStartYear%wasSet = .FALSE.
+                IF (CurArgs >= 14) THEN
+                  IF (TRIM(InArgs(14)) /= Blank) THEN
+                    RunPeriodStartYear%wasSet = .TRUE.
+                    RunPeriodStartYear%originalValue = InArgs(14)
+                  END IF
+                END IF
+                RunPeriodRepeated%wasSet = .FALSE.
+                IF (CurArgs >= 12) THEN
+                  IF (TRIM(InArgs(12)) /= Blank) THEN
+                    RunPeriodRepeated%wasSet = .TRUE.
+                    RunPeriodRepeated%originalValue = InArgs(12)
+                  END IF
+                END IF
+                ! Now start writing some object data out
+                ! Name, BeginMonth and BeginDay are the same
+                OutArgs(1:3) = InArgs(1:3) 
+                ! Start year is weird
+                IF (RunPeriodStartYear%wasSet) THEN
+                  OutArgs(4) = RunPeriodStartYear%originalValue
+                ELSE IF (RunPeriodRepeated%wasSet) THEN
+                  READ(OutArgs(2), *) MonthNumber
+                  READ(OutArgs(3), *) DayNumber
+                  IF (TRIM(InArgs(6)) /= Blank) THEN
+                    YearNumber = FindYearForWeekDay(MonthNumber, DayNumber, InArgs(6))
+                  ELSE
+                    YearNumber = FindYearForWeekDay(MonthNumber, DayNumber, "SUNDAY")
+                  END IF
+                  WRITE(OutArgs(4), *) YearNumber
+                ELSE
+                  OutArgs(4) = Blank
+                END IF
+                ! End month and End day or month are the same, just need to shift one field
+                OutArgs(5:6) = InArgs(4:5)
+                ! End year is weird
+                IF (RunPeriodRepeated%wasSet) THEN
+                  ! What if start year turns out to be blank above?
+                  IF (TRIM(OutArgs(4)) == Blank) THEN
+                    OutArgs(7) = Blank
+                  ELSE
+                    READ(RunPeriodRepeated%originalValue, *) RepeatedCount
+                    READ(OutArgs(4), *) YearNumber
+                    YearNumber = YearNumber + RepeatedCount
+                    WRITE(OutArgs(7), *) YearNumber
+                    IF (TRIM(InArgs(4))=="2".AND.TRIM(InArgs(5))=="29") THEN
+                      ! We should have a leap year end year
+                      IF (.NOT.IsYearNumberALeapYear(YearNumber)) THEN
+                        ! Warning about bad end year/end date combination
+                        OutArgs(6) = "28"
```

Comment:
Several places have a comment about a warning, but I don't see any code to throw one. Or do you mean EnergyPlus 9.0 will throw the warning?